Fix blob reading in SQLServer

markstory opened the pull request:
VARBINARY fields in SQLServer come back as hexadecimal encoded strings. In order for these to be useful, they need to be converted into resources like other blobby things.

This should fix that AppVeyor builds as well.

markstory committed:
Fix blob reading in SQLServer

VARBINARY fields in SQLServer come back as hexadecimal encoded strings.
In order for these to be useful, they need to be converted into
resources like other blobby things.

This should fix that AppVeyor builds as well.

lorenzo commented on src/Database/Type/BinaryType.php:
```diff
@@ -55,6 +55,9 @@ public function toPHP($value, Driver $driver)
         if ($value === null) {
             return null;
         }
+        if (is_string($value) && preg_match('/^[a-zA-Z0-9]+$/', $value)) {
```
I think I would prefer checking explicitly if driver is SQLServer rather than running a preg on a potential couple hundred megabyte strings.

markstory replied:
Ok, I can do that.

markstory replied:
Actually testing the type will require loading the Sqlserver driver class, which we can't do as PDO constants are only defined when pdo_sqlsrv is installed.

markstory replied:
I found a way around the constant issue, hopefully its not too hacky.

markstory committed:
Replace regex with instanceof check.

The hexadecimal check could get very expensive with big blobs. Instead
we can sniff out the Sqlserver class and change behavior based on that.

In order for the SQLServer driver to be loadable in environments where
pdo_sqlsrv is not installed a constant needed to be replaced with an
integer.

lorenzo commented:
That was clever :D

markstory commented:
This also means we can unskip a bumch of the sqlserver tests in non windows environments which will be nice.
